src/OpenColorIO/Config.cpp

```diff
@@ -1078,6 +1094,435 @@ class Config::Impl
+                        {
+                            refSpace = builtinLinearSpaces[i];
+                        }
+
+                        return refSpace;
+                    }
+                }
+            }
+        }
+
+        return "";
+    }
+
+    std::string checkForSRGBTextureColorSpace(ConstConfigRcPtr & config, 
+                                             ConstColorSpaceRcPtr cs,
+                                             ConstConfigRcPtr & builtinConfig,
+                                             std::vector<std::string> & builtinLinearSpaces) const
+    {
+        // If the color space is an sRGB texture space, return the reference space used by the config.
+
+        // Get a transform in the to-reference direction.
+        ConstTransformRcPtr toRefTransform;
+        ConstTransformRcPtr ctransform = cs->getTransform(COLORSPACE_DIR_TO_REFERENCE);
+        if (ctransform) 
+        {
+            toRefTransform = ctransform;
+        }
+        else
+        {
+            ctransform = cs->getTransform(COLORSPACE_DIR_FROM_REFERENCE);
+            if (ctransform)
+            {
+                TransformRcPtr transform = ctransform->createEditableCopy();
+                transform->setDirection(TRANSFORM_DIR_INVERSE);
+                toRefTransform = transform;
+            }
+            else
+            {
+                // Both directions missing.
+                return "";
+            }
+        }
+
+        // First check if it has the right non-linearity. The objective is to fail quickly on color 
+        // spaces that are definitely not sRGB before proceeding to the longer test of guessing the 
+        // reference space primaries.
+
+        // Break point is at 0.039286, so include at least one value below this.
+        std::vector<float> vals =
+        {
+            0.5f,  0.5f,  0.5f, 
+            0.03f, 0.03f, 0.03f, 
+            0.25f, 0.25f, 0.25f, 
+            0.75f, 0.75f, 0.75f, 
+            0.f,   0.f,   0.f, 
+            1.f,   1.f ,  1.f
+        };
+        std::vector<float> out = vals;
+
+        PackedImageDesc desc(&vals[0], (long) vals.size()/3, 1, CHANNEL_ORDERING_RGB);
+        PackedImageDesc descDst(&out[0], (long) vals.size()/3, 1, CHANNEL_ORDERING_RGB);
+
+        ConstProcessorRcPtr proc  = getProcessorWithoutCaching(*config,
+                                                               toRefTransform,
+                                                               TRANSFORM_DIR_FORWARD);
+
+        ConstCPUProcessorRcPtr cpu  = proc->getOptimizedCPUProcessor(OPTIMIZATION_LOSSLESS);
+        cpu->apply(desc, descDst);
+
+        for (size_t i = 0; i < out.size(); i++)
+        {
+            // Apply the sRGB function (linear to non-lin).
+            if (out[i] <= 0.0030399346397784323f)
+            {
+                out[i] *= 12.923210180787857f;
+            }
+            else
+            {
+                out[i] = 1.055f * pow(out[i], 1/2.4f) - 0.055f;
```

Comment:
I think we should use `std::pow` instead of bare `pow` from earlier comments by @hodoulp.

Reply:
Will change to std::pow